### PR TITLE
chore(more log): add more logs for getting the cause of the error better

### PR DESCRIPTION
**Pull request by tigercl**
After this PR is merged, a new tag needs to be marked.

### CODE REVIEW

**Review by Rory-Z — APPROVED** [review]

**Review by terry-xiaoyu — APPROVED** [review]

File: src/ehttpc.erl [hunk 1]
```diff
@@ -209,6 +210,7 @@ handle_info({gun_down, Client, _, Reason, KilledStreams, _}, State = #state{clie
     {noreply, State#state{gun_state = down, requests = NRequests}};
 
 handle_info({'DOWN', MRef, process, Client, Reason}, State = #state{mref = MRef, client = Client, requests = Requests}) ->
+    ?LOG(warning, "The process of gun exited due to ~p", [Reason]),
```
Comment (zmstone) [inline src/ehttpc.erl:213]:
```suggestion
    ?LOG(warning, "gun_process_exited due to ~p", [Reason]),
```

File: src/ehttpc.erl [hunk 2]
```diff
@@ -194,6 +194,7 @@ handle_info({gun_up, Client, _}, State = #state{client = Client}) ->
     {noreply, State#state{gun_state = up}};
 
 handle_info({gun_down, Client, _, Reason, KilledStreams, _}, State = #state{client = Client, requests = Requests}) ->
+    ?LOG(warning, "Received gun_down with ~p", [Reason]),
```
Comment (zmstone) [inline src/ehttpc.erl:197]:
```suggestion
    ?LOG(warning, "received_gun_down with ~p", [Reason]),
```

Comment (zmstone) [inline src/ehttpc.erl:197]:
```suggestion
    ?LOG(warning, "received_gun_down with ~p", [Reason]),
```

File: src/ehttpc.erl (hunk 1 above)
Comment (tigercl) [reply src/ehttpc.erl:213]: I think it is more reasonable to appear as a normal statement in the log, and adding an underscore will make it look like the information returned from a certain function.

**Pull request merged**